build/tools/make-winpe-i.MX8.cmd

```diff
@@ -0,0 +1,234 @@
+    exit /b 1
+)
+
+if "%UEFI_PATH%" == "" (
+    echo Missing required option '/uefi'. Type /? for usage.
+    exit /b 1
+)
+
+:: make temporary WinPE directory
+rmdir /s /q "%DEST%" > NUL 2>&1
+mkdir "%DEST%"
+mkdir "%DEST%\sources"
+
+:: copy firmware files
+echo Copying firmware from %FIRMWARE_PATH%
+copy "%FIRMWARE_PATH%" "%DEST%" /y || goto err
+
+echo Copying UEFI from %UEFI_PATH%
+copy "%UEFI_PATH%" "%DEST%\uefi.fit" /y || goto err
+
+:: copy winpe files
+echo Creating WinPE image layout at %DEST%
+echo Copying WinPE files from %WINPE_DIR%
+if not exist "%WINPE_DIR%\Media" (
+    echo You must install the Windows PE Add-on for the ADK
+    echo https://docs.microsoft.com/en-us/windows-hardware/manufacture/desktop/download-winpe--windows-pe
+    goto err
+)
+xcopy /herky "%WINPE_DIR%\Media" "%DEST%\" || goto err
+copy "%WINPE_DIR%\en-us\winpe.wim" "%DEST%\sources\boot.wim" || goto err
+move "%DEST%\bootmgr.efi" "%DEST%\EFI\Microsoft\boot\" || goto err
+
+:: BCD
+echo Enabling kernel debugging
+set TARGET_BCD_STORE=%DEST%\EFI\Microsoft\boot\bcd
+bcdedit /store "%TARGET_BCD_STORE%" /dbgsettings SERIAL DEBUGPORT:1 BAUDRATE:115200 || goto err
+bcdedit /store "%TARGET_BCD_STORE%" /debug {default} on || goto err
+
+echo Enable boot test/flight signing
+bcdedit /store "%TARGET_BCD_STORE%" /set {bootmgr} flightsigning on || goto err
+bcdedit /store "%TARGET_BCD_STORE%" /set {bootmgr} testsigning on || goto err
+
+echo Enable kernel test/flight signing...
+bcdedit /store "%TARGET_BCD_STORE%" /set {default} testsigning on || goto err
+bcdedit /store "%TARGET_BCD_STORE%" /set {default} flightsigning on || goto err
+
+mkdir "%MOUNT_DIR%" > NUL 2>&1
+echo Mounting WIM at %MOUNT_DIR%
+dism /mount-wim /wimfile:"%DEST%\sources\boot.wim" /mountdir:"%MOUNT_DIR%" /index:1 || goto err
+
+set STARTNET_CMD=%MOUNT_DIR%\Windows\System32\startnet.cmd
+if NOT "%FFU_PATH%" == "" (
+    echo Setting up FFU deployment to MMC
+    copy "%FFU_PATH%" "%DEST%\Flash.ffu"
+
+    echo Appending FFU flashing commands to %STARTNET_CMD%
+    echo echo Deploying Flash.ffu to eMMC >> "%STARTNET_CMD%"
+    echo dism /Apply-Image /ImageFile:%WINPE_DRIVE_LETTER%:\Flash.ffu /ApplyDrive:\\.\PhysicalDrive%FFU_DISK_NUM% /skipplatformcheck >> "%STARTNET_CMD%"
+    echo rmdir /s /q %WINPE_DRIVE_LETTER%:\_efi >> "%STARTNET_CMD%"
+    echo ren %WINPE_DRIVE_LETTER%:\efi _efi >> "%STARTNET_CMD%"
+    echo echo Successfully flashed new image to eMMC. Restarting the device... >> "%STARTNET_CMD%"
+    echo wpeutil reboot >> "%STARTNET_CMD%"
+)
+
+echo Copying driver packages to %PACKAGES_DIR%
+call :GETPACKAGE NXP.iMX.HalExtiMXDma.cab || goto err
+call :GETPACKAGE NXP.iMX.imxecspi.cab || goto err
+call :GETPACKAGE NXP.iMX.imxgpio.cab || goto err
+call :GETPACKAGE NXP.iMX.imxi2c.cab || goto err
+call :GETPACKAGE NXP.iMX.imxnetmini.cab || goto err
+call :GETPACKAGE NXP.iMX.imxuart.cab || goto err
+call :GETPACKAGE NXP.iMX.imxusdhc.cab || goto err
+

```

Comment:
For what reason did you remove add the NXP.iMX.Optee.Tree cab for the iMX8?

Reply:
I thought it wasn't used. I added it back.